raillabel_providerkit/validation/validate_dimensions/_dimensions.py

```diff
@@ -0,0 +1,218 @@
+# Copyright DB InfraGO AG and contributors
+# SPDX-License-Identifier: Apache-2.0
+
+from dataclasses import dataclass
+
+from raillabel.format import Cuboid
+
+
+@dataclass
+class _DimensionRange:
+    min: float
+    max: float
+
+
+@dataclass
+class _TypeDimensions:
+    type: str | list[str]
+    attributes: dict[str, str | list[str]] | None = None
+    height: _DimensionRange | None = None
+    width: _DimensionRange | None = None
+
+    def applies(self, type_: str, annotation: Cuboid) -> bool:
+        """Return True if this _TypeDimensions instance applies to the given annotation."""
+        self._preprocess_variables()
+
+        if type_ not in self.type:
+            return False
```

Comment:
When `type` is a `str`, this would also pass if `type_` is a substring of `type`. Perhaps we should catch that just in case?
```suggestion
        if (isinstance(self.type, str) and type_ != self.type) or (
            isinstance(self.type, list) and type_ not in self.type
        ):
            return False
```

Reply:
This is avoided by preprocessing the variables in line 42-43.